core/templates/pages/exploration-editor-page/editor-tab/unresolved-answers-overview/unresolved-answers-overview.component.spec.ts

```diff
@@ -0,0 +1,182 @@
+  // eslint-disable-next-line max-len
+  'components/state-editor/state-editor-properties-services/state-interaction-id.service';
+
+describe('Unresolved Answers Overview Component', function() {
+  var $q = null;
+  var $rootScope = null;
+  var $scope = null;
+  var $uibModal = null;
+  var editabilityService = null;
+  var explorationStatesService = null;
+  var improvementsService = null;
+  var stateInteractionIdService = null;
+  var stateEditorService = null;
+  var stateTopAnswersStatsService = null;
+
+  var stateName = 'State1';
+
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(function() {
+    editabilityService = TestBed.get(EditabilityService);
+    stateInteractionIdService = TestBed.get(StateInteractionIdService);
+  });
+
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    $q = $injector.get('$q');
+    $rootScope = $injector.get('$rootScope');
+    $uibModal = $injector.get('$uibModal');
+    explorationStatesService = $injector.get('ExplorationStatesService');
+    improvementsService = $injector.get('ImprovementsService');
+    stateEditorService = $injector.get('StateEditorService');
+    stateTopAnswersStatsService = $injector.get('StateTopAnswersStatsService');
+
+    spyOn(stateEditorService, 'getActiveStateName').and.returnValue(stateName);
+    spyOn(explorationStatesService, 'getState').and.returnValue({});
+
+    $scope = $rootScope.$new();
+    var ctrl = $componentController('unresolvedAnswersOverview', {
+      $rootScope: $rootScope,
+      $scope: $scope,
+      EditabilityService: editabilityService,
+      StateInteractionIdService: stateInteractionIdService
+    });
+    ctrl.$onInit();
+  }));
+
+  it('should evaluate $scope properties after controller initialization',
+    function() {
+      expect($scope.unresolvedAnswersOverviewIsShown).toBe(false);
+      expect($scope.SHOW_TRAINABLE_UNRESOLVED_ANSWERS).toBe(false);
+    });
+
+  it('should check when unresolved answers overview are shown', function() {
+    spyOn(stateTopAnswersStatsService, 'hasStateStats').and.returnValue(true);
+    spyOn(
+      improvementsService,
+      'isStateForcedToResolveOutstandingUnaddressedAnswers')
+      .and.returnValue(true);
+
+    expect($scope.isUnresolvedAnswersOverviewShown()).toBe(true);
+  });
+
+  it('should check when unresolved answers overview are not shown because it' +
+    ' has no state stats', function() {
+    spyOn(stateTopAnswersStatsService, 'hasStateStats').and.returnValue(false);
+    spyOn(
+      improvementsService,
+      'isStateForcedToResolveOutstandingUnaddressedAnswers');
+
+    expect($scope.isUnresolvedAnswersOverviewShown()).toBe(false);
+    expect(
+      improvementsService.isStateForcedToResolveOutstandingUnaddressedAnswers)
+      .not.toHaveBeenCalled();
+  });
+
+  it('should check when unresolved answers overview are not shown because' +
+    ' there is no state foced to resolve', function() {
```

Comment:
```suggestion
  it('should check that unresolved answers overview are not shown because' +
    ' the state is not forced to resolve unaddressed answers', function() {
```

Reply:
Done.